Native.xs

```diff
@@ -421,9 +422,11 @@ _getaddrinfo(Net_DNS_Native *self, char *host, SV* sv_service, SV* sv_hints, int
             DNS_reinit_pool(self);
         }
 #endif
-        if (socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, PF_UNSPEC, fd) != 0)
+        if (socketpair(AF_UNIX, SOCK_STREAM, PF_UNSPEC, fd) != 0)
             croak("socketpair(): %s", strerror(errno));
-        
+
+        fcntl(fd, F_SETFD, FD_CLOEXEC);
```

Comment:
Does it even compile? `fd` is `int[]`, but `fcntl()` accepts `int`

Reply:
It compiled, passed the tests, and installed successfully on both MacOS 10.13 and Fedora 29.

Reply:
Although it did warn about it on Fedora 29. Fixing it now…